spec/tests/requires.fmf

```diff
@@ -0,0 +1,29 @@
+summary: Packages required for the test execution
+
+story:
+    As a tester I want to specify packages which are required by
+    the test and need to be installed on the system so that the
+    test can be successfully executed.
+
+description: |
+    In order to execute the test, additional packages may need to
+    be installed on the system. For example `gcc` and `make` are
+    needed to compile tests written in C on the target machine. If
+    the package cannot be installed test execution should result
+    in an ``error``.
+
+    Packages being tested (system under test) should not be listed
+    here. Instead, they should be covered by plans (L2 metadata).
+    This especially applies when there are multiple packages
+    covered by a single test (regular package, module, software
+    collection) or the component has different names in different
+    product versions.
+
+    Should be a ``list of strings``.
```

Comment:
Forgive me opening can of worms: What if `requires` are different between underlying distribution (RHEL-6 vs RHEL-8 with modules). To give an example: command `ps` is part of `procps-ng` since rhel-7. However for rhel-6 it is in `procps` package. What would be the correct way? 
- Have 'rhel-6' special rhel-6 virtual test case ? 
- Leave such conflicts for L2 metadata ?

Reply:
Or use binary as a dependency? `[ /usr/bin/ps ]`

Reply:
I think for such cases cases there are two solutions:

* set these per-distro special settings in L2 metadata (natural place)
* don't share test cases across distros and store requires in L1 metadata

Reply:
> Or use binary as a dependency? `[ /usr/bin/ps ]`

Good idea. Just relevant note: Providing a binary path does not work with ansible, right?

Reply:
@psss, I fail to see how ansible fits in here. Could you elaborate?

The steps' `requires` are handled currently without ansible, is there any reason to change that?

Reply:
If you move the setup to L2 you would probably have everything related to the environment setup stored at one place which could naturally be an ansible playbook. Then using binary paths would not be an option as it is currently not supported. You would have to workaround that by additional section in the prepare step.

Reply:
In worst case (if Ansible decides not to support binaries) you can run dnf/yum directly via Ansible shell. But IMHO that is task for the harness to figure out the way how to install requires.

What exactly can represent strings in the `requires`? 
- Package name for sure. 
- Binary path (`/usr/bin/pip3`)?
- Provides (`python3dist(pip)`)?

Reply:
Well, we could possibly explicitly define that the `requires` field will be implemented using `dnf` which would resolve the question above (e.g. you can use any package specification supported by dnf) as well as the problem with ansible not supporting binary paths. Does that make sense?

Reply:
Included the following clarification which should resolve this:

    Should be a ``list of strings`` using package specification
    supported by ``dnf`` which takes care of the installation.